src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PETypeParameterSymbol.cs

```diff
@@ -234,31 +254,44 @@ public override bool HasConstructorConstraint
         {
             get
             {
-                return (_flags & GenericParameterAttributes.DefaultConstructorConstraint) != 0;
+                GetDeclaredConstraintTypes();
+                return (_lazyFlags & GenericParameterAttributes.DefaultConstructorConstraint) != 0;
             }
         }
 
         public override bool HasReferenceTypeConstraint
         {
             get
             {
-                return (_flags & GenericParameterAttributes.ReferenceTypeConstraint) != 0;
+                GetDeclaredConstraintTypes();
```

Comment:
>GetDeclaredConstraintTypes(); [](start = 16, length = 29)

Given my comments for GetDeclaredConstraintTypes, I don't think this change is necessary